src/components/preset-control/preset-control.tsx

```diff
@@ -0,0 +1,193 @@
+                isPopoverOpen && styles.presetControlButton_open,
+              )}
+            />
+          )}
+        </span>
+      </button>
+      <Popover
+        isOpen={isPopoverOpen}
+        onClose={closePopover}
+        anchorEl={popoverAnchor}
+      >
+        <section className={styles.presetControlCard}>
+          <header className={styles.presetControlCard__header}>
+            <button
+              disabled={isCreateDisabled}
+              className={cn(
+                styles.presetControlCard__createButton,
+                isCreateDisabled &&
+                  styles.presetControlCard__createButton_disabled,
+              )}
+              onClick={() => {
+                closePopover()
+                openCreateDialog()
+              }}
+            >
+              {t('presetControl.createNewPreset')}
+            </button>
+          </header>
+          {presets && (
+            <PresetControlList
+              className={styles.presetControlCard__list}
+              presets={presets}
+              selected={selected}
+              onSelect={setSelected}
+              onModify={presetName => {
+                closePopover()
+                onModify(presetName)
+              }}
+              onDelete={presetName => {
+                closePopover()
+                openDeleteDialog({ presetName })
+              }}
+            />
+          )}
+          <footer className={styles.presetControlCard__actions}>
+            <Button
+              className={styles.presetControlCard__button}
+              variant="tertiary"
+              text={t('general.cancel')}
+              onClick={closePopover}
+            />
+            {onJoin && (
+              <Button
+                className={styles.presetControlCard__button}
+                variant="secondary"
+                text={t('presetControl.join')}
+                disabled={!selected}
+                onClick={() => {
+                  onJoin(selected)
+                  closePopover()
+                }}
+              />
+            )}
+            <Button
+              className={styles.presetControlCard__button}
+              text={t('presetControl.apply')}
+              disabled={!selected}
+              onClick={() => {
+                onApply(selected)
+                closePopover()
+              }}
+            />
+          </footer>
+        </section>
+      </Popover>
+      <PresetDeleteDialog
+        {...deleteDialog}
+        onClose={closeDeleteDialog}
+        onDelete={() => {
+          closeDeleteDialog()
+          if (deleteDialog.presetName) {
+            onDelete(deleteDialog.presetName)
+          }
+        }}
+        presetName={selected}
+      />
+      <PresetCreateDialog
+        {...createDialog}
+        onClose={closeCreateDialog}
+        onCreate={presetName => {
+          closeCreateDialog()
+          onCreate(presetName)
+        }}
+      />
+    </Fragment>
```

Comment:
Too long markup section, try to split on dumb components or render functions

Reply:
fixed